src/main/java/me/ajaja/module/tag/adapter/out/persistence/CreateTagAdapter.java

```diff
@@ -0,0 +1,52 @@
+package me.ajaja.module.tag.adapter.out.persistence;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import lombok.RequiredArgsConstructor;
+import me.ajaja.module.tag.application.port.in.CreateTagUseCase;
+import me.ajaja.module.tag.application.port.out.CreateTagPort;
+import me.ajaja.module.tag.domain.FootprintTag;
+import me.ajaja.module.tag.domain.PlanTag;
+import me.ajaja.module.tag.domain.Tag;
+
+@Repository
+@Transactional
+@RequiredArgsConstructor
+public class CreateTagAdapter implements CreateTagPort {
+	private final TagRepository tagRepository;
+	private final FootprintTagRepository footprintTagRepository;
+	private final PlanTagRepository planTagRepository;
+
+	@Override
+	public void create(CreateTagUseCase.Type type, Long targetId, List<String> tagNames) {
+		Set<String> tagNameSet = new LinkedHashSet<>(tagNames);
+		List<Tag> savedTags = tagNameSet.stream().map(this::getOrCreateTagIfNotExists).toList();
+
+		switch (type) {
+			case FOOTPRINT -> saveFootprintTag(targetId, savedTags);
+			case PLAN -> savePlanTag(targetId, savedTags);
+		}
+	}
+
+	private Tag getOrCreateTagIfNotExists(String name) {
+		return tagRepository.findByName(name)
+			.orElseGet(() -> tagRepository.save(new Tag(name)));
+	}
+
+	private void saveFootprintTag(Long footprintId, List<Tag> savedTags) {
+		savedTags.stream().map(tag -> new FootprintTag(footprintId, tag.getId())).forEach(
+			footprintTag -> footprintTagRepository.save(footprintTag)
+		);
```

Comment:
반복을 두번(`map()`에서 한번, `forEach()`에서 한번) 돌릴 필요가 없을 것 같은데요??
```suggestion
		for (Tag tag: savedTags) {
		    footprintTagRepository.save(footprintId, tag.getId());
		}
```

Reply:
아니면 차라리 이렇게 하는게 나아 보입니당!
```suggestion
		List<FootprintTag> footprintTags = savedTags.stream()
		.map(tag -> new FootprintTag(footprintId, tag.getId()))
		.toList();

		footprintTagRepository.saveAll(footprintTags);
```

Reply:
마지막으로 제시해주신게 가독성 측면에서  깔끔해서 수정했습니다.

하지만 성능적으로는 차이가 없다고 생각이 듭니다. 
제가 잘못아는 것일 수 있어서 가독성 이외의 의도가 있다면 이야기 해보면 좋겠습니다.

- 첫번째 방법 
반복을 두번 돌린다기 보단 하나의 스트림 (위) 가 반복문(아래) 이것 동일하다고 봅니다. 
new 연산을 해야하기도 하고요.
```
.map(tag -> new FootprintTag(footprintId, tag.getId())).forEach(
			footprintTag -> footprintTagRepository.save(footprintTag)

``` 
` footprintTagRepository.save(footprintId, tag.getId());`

- 두번째 방법
상위 트랜잭션에서 save 반복문과 saveAll 은 동일하다고 생각합니다.

Reply:
개인적으로 첫번째 방법이 더 깔끔했던 것 같아요!!
두번째 방법도 말 그대로 결국 내부적으로 반복을 한번 더 돌리는 거니까요!

그리고 저 스트림을 사용하는 형태도 엄연히 반복이 두번 도는 것 아닌가욤??
map에서 한번씩 돌아가면서 `Stream<FootprintTag>`를 만들고 그걸 받아서 `forEach()`로 돌리는 거잖아요!

Reply:
피드백 반영 : c77324a043886c480e74b10ee3b94d872f721597

맞네요 생각해보니깐 두번쨰 한번 더 돌리네요.
전 스트림이 한번에 쭉가서 . . 지나서 크게 한번씩 도는줄 알았어요.

근데 다 떠나서 첫번쨰가 더 간결하고 실제 속도도 빨라요1